src/main/scala/za/co/absa/abris/examples/ConfluentKafkaAvroWriter.scala

```diff
@@ -18,88 +18,64 @@ package za.co.absa.abris.examples
 import za.co.absa.abris.examples.data.generation.ComplexRecordsGenerator
-import za.co.absa.abris.examples.utils.ExamplesUtils._
-
-import scala.collection.JavaConverters._
 
 
 object ConfluentKafkaAvroWriter {
 
-  private val PARAM_JOB_NAME = "job.name"
-  private val PARAM_JOB_MASTER = "job.master"
-  private val PARAM_PAYLOAD_AVRO_SCHEMA = "payload.avro.schema"
-  private val PARAM_KEY_AVRO_RECORD_NAME = "avro.key.record.name"
-  private val PARAM_KEY_AVRO_RECORD_NAMESPACE = "avro.key.record.namespace"
-  private val PARAM_VALUE_AVRO_RECORD_NAME = "avro.value.record.name"
-  private val PARAM_VALUE_AVRO_RECORD_NAMESPACE = "avro.value.record.namespace"
-  private val PARAM_INFER_SCHEMA = "infer.schema"
-  private val PARAM_LOG_LEVEL = "log.level"
-  private val PARAM_TEST_DATA_ENTRIES = "test.data.entries"
-  private val PARAM_EXECUTION_REPEAT = "execution.repeat"
-  private val PARAM_NUM_PARTITIONS = "num.partitions"
-  private val PARAM_TOPIC = "option.topic"
-
-  def main(args: Array[String]): Unit = {
-
-    // there is a sample properties file at /src/test/resources/DataframeWritingExample.properties
-    checkArgs(args)
+  val kafkaTopicName = "test_topic"
 
-    val properties = loadProperties(args)
+  val dummyDataRows = 5
+  val dummyDataPartitions = 1
 
-    val spark = getSparkSession(properties, PARAM_JOB_NAME, PARAM_JOB_MASTER, PARAM_LOG_LEVEL)
+  def main(args: Array[String]): Unit = {
 
-    spark.sparkContext.setLogLevel(properties.getProperty(PARAM_LOG_LEVEL))
+    val spark = SparkSession
+      .builder()
+      .appName("ReaderJob")
+      .master("local[2]")
+      .getOrCreate()
 
-    import spark.implicits._
+    spark.sparkContext.setLogLevel("INFO")
 
-    implicit val encoder: Encoder[Row] = getEncoder
+    val dataFrame = generateRandomDataFrame(spark)
 
-    do {
-      val rows = createRows(properties.getProperty(PARAM_TEST_DATA_ENTRIES).trim().toInt)
-      val dataFrame = spark.sparkContext.parallelize(rows, properties.getProperty(PARAM_NUM_PARTITIONS).toInt).toDF()
+    dataFrame.show(false)
 
-      dataFrame.show(false)
+    val schemaString = ComplexRecordsGenerator.usedAvroSchema
 
-      toAvro(dataFrame, properties.asScala.toMap) // check the method content to understand how the library is invoked
-        .write
-        .format("kafka")
-        .addOptions(properties) // 1. this method will add the properties starting with "option."
-        .save()                 // 2. security options can be set in the properties file
-    } while (properties.getProperty(PARAM_EXECUTION_REPEAT).toBoolean)
-  }
+    // to serialize all columns in dataFrame we need to put them in a spark struct
+    val allColumns = struct(dataFrame.columns.head, dataFrame.columns.tail: _*)
 
-  private def toAvro(dataFrame: Dataset[Row], properties: Map[String, String]) = {
+    val abrisConfig = AbrisConfig
+      .toConfluentAvro
+      .provideAndRegisterSchema(schemaString)
+      .usingTopicNameStrategy(kafkaTopicName)
```

Comment:
Interestingly, on the reader example it is `andTopicNameStrategy`. Is it intentional?

Reply:
Yes it's intentional. If you read it should make normal sentence. 

- downloadReaderSchemaByLatestVersion(...).andTopicNameStrategy(...) = 
download shcema **By LatestVersion and TopicNameStrategy**

- provideAndRegisterSchema(...).usingTopicNameStrategy(...) = 
register schema **using TopicNameStrategy**

The difference is that when schema is donwloaded is identified by version and strategy, but when you register you just use the strategy to register